### DISCUSSION TITLE
How to downgrade chia code?

### DISCUSSION THREAD

**Discussion by af85** [Chia-Network/chia-blockchain#557] [Q&A] [answered]
How do  I downgrade b18 to b17?

Normally, I upgrade with the following procedure:
1. spin down plots
2. stop chia processes
3. rename chia git directory
4. git clone new chia
5. sh install, activate, init

Luckily I still have my b17 directory, so I did the following:
1. kill all plots and clear temp directories
2. stop chia processes
3. rm -rf ~/.chia/beta-1.0b18
3. mv b17 git backup to chia git dir
4. activate, init
5. run gui, full node, etc
6. chia plots create

So my questions:
1. Is my downgrade procedure safe? The `sh install` i did when upgrading to b18 upgraded several python packages it seems. I want to make sure there won't be issues with b17.
2. When I was running b18, I was up to ~1100 TXCH, but now i'm down to what I had when I was on b17. Why did I lose them? Was there something in my ~/.chia/beta-1.0b18/ directory that had my old test coins?

I think there should be a "how to downgrade" or "how to install a specific version" section in the wiki. I pulled the tarball for b17, and got some error about missing files due to it not being a git directory.

**Comment by hoffmang9** [comment, marked as answer]
The easiest way is to go into whichever git cloned chia-blockchain directory and type:
`git checkout 1.0beta17; rm -rf venv; sh install.sh ; . ./activate`

Your coins are on the chain - not on your disk. The wallet db may be confused. Delete it and resync it. It's in `~/.chia/VERSION/wallet/db/`
